src/geometry/Lie_group/lie_group.lean

```diff
@@ -0,0 +1,224 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import geometry.manifold.constructions
+
+noncomputable theory
+
+/-!
+# Lie groups
+
+We define Lie groups.
+
+## Main definitions and statements
+
+* `Lie_add_group I G` : a Lie additive group where `G` is a manifold on the model with corners `I`.
+* `Lie_group I G`     : a Lie multiplicative group where `G` is a manifold on the model with
+                        corners `I`.
+
+## Implementation notes
+A priori, a Lie group here is a manifold with corner.
```

Comment:
What does this buy you? You won't get any example with boundary, this would violate homogeneity, right? Can't you simplify stuff by restricting to models without boundary?

Reply:
This is exactly what I did in when I first wrote Lie groups but Sébastien advised against it so that is why I changed it. I can tell you his reasons but he probably can explain to you better himself!

Reply:
Models without boundary are a particular case of models with boundary, so I don't really see what you would like to do. One thing you can try is to require that the model with corners is the identity model with corners on a vector space, but this is not stable under product, so the product of two Lie groups would not be a Lie group if you required this.

One thing you can do is to require the additional assumption `[boundaryless I]` which is the way we have to say in a canonical way that there is no boundary. And indeed if it turns out to be useful at some point, we will definitely add it, either to the statements that need it or to the definition of Lie group. But I don't think this has shown up yet.

You could maybe add a comment about this in some docstring?

Reply:
Is you in the first paragraph referred to me or to Patrick?

Reply:
It's Patrick, sorry for being unclear. Or a generic "you".